enterprise/admin/views.py

```diff
@@ -241,20 +244,23 @@ def get_search_keyword(self, request):
         """
         return request.GET.get('q', None)
 
-    def get_enterprise_customer_user_queryset(self, search_keyword, customer_uuid):
+    def get_enterprise_customer_user_queryset(self, request, search_keyword, customer_uuid, page_size=10):
         """
         Get the list of EnterpriseCustomerUsers we want to render.
 
-        Args:
+        Arguments:
+            request (HttpRequest): HTTP Request instance.
             search_keyword (str): The keyword to search for in users' email addresses and usernames.
             customer_uuid (str): A unique identifier to filter down to only users linked to a
             particular EnterpriseCustomer.
+            page_size (int): Number of learners displayed in each paginated set.
         """
-        # TODO: Add paging so the admin view does not time out.
-        # For now, we will allow search, but will return an empty list
-        # if no search keyword is provided.
-        learners = []
-        if search_keyword is not None:
+        page = request.GET.get('page', 1)
+        show_all = request.GET.get('all', None)
+
+        if show_all or search_keyword is None:
+            learners = EnterpriseCustomerUser.objects.all()
```

Comment:
Why we need this `if` block? As per my understanding we only need the code inside below `else` block plus `learners = paginated_list(learners, page, page_size)`